pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -122,19 +129,57 @@ func verifyDevicePath(diskUUID string) string {
 	return ""
 }
 
+// scans storage devices and adds path in dev/disk/by-id
 func probeAttachedVolume() error {
-	executor := exec.New()
-	args := []string{"trigger"}
-	cmd := executor.Command("/usr/bin/udevadm", args...)
-	_, err := cmd.CombinedOutput()
+	var (
+		scsiPath   = "/sys/class/scsi_host/"
+		driverType = "mpt"
+		scanVal    = []byte("- - -")
+	)
+
+	files, err := ioutil.ReadDir(scsiPath)
 	if err != nil {
-		glog.Errorf("error running udevadm trigger %v\n", err)
-		return err
+		glog.V(2).Infof("Error scanning attached vSphere volumes: %v", err)
+		return ErrProbeVolume
+	}
+	for _, file := range files {
+		driverFile := scsiPath + file.Name() + "/proc_name"
+
+		// grep for 'mpt' in 'proc_name'
+		if total := grepFile(driverFile, []byte(driverType)); total != 0 {
```

Comment:
another thought, what if you just scan all scsi bus as we did for fibre channel [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/fc/fc_util.go#L108)